lib/facter/puppetdb_version.rb

```diff
@@ -2,7 +2,26 @@
   confine { Facter::Util::Resolution.which('puppetdb') }
 
   setcode do
-    output = Facter::Core::Execution.execute('puppetdb --version')
-    output.split(':').last.strip
+    require 'open3'
+
+    # check if os is debian/ubuntu and the package is not from puppetlabs
+    if Facter.value(:osfamily) == 'Debian'
+      package_maintainer = Facter::Core::Execution.execute('apt-cache show puppetdb | grep "Maintainer:" | head -1')
+      unless package_maintainer.include? 'Puppet Labs'
+        output, status = Open3.capture2('dpkg-query --showformat=\'${Version}\' --show puppetdb')
```

Comment:
is there a reason for using `Open3` here instead of `Facter::Core::Execution.execute`?

Reply:
so i can do `status.success?` and react if it does not work.